Prior auth

juggernautsei opened the pull request:
Here is an update the prior auth form to make it a little more usable than the previous prior auth form.

bradymiller commented on contrib/forms/prior_auth_behavior_health/README-FIRST.txt:
```diff
@@ -0,0 +1,8 @@
+
```
If you are modifying the original prior auth forms, then should instead place all these changes in contrib/forms/prior_auth/

bradymiller replied:
disregard above comment for now. First need to figure out if this needs to be a new form or not.

bradymiller commented on interface/forms/prior_auth/C_FormPriorAuth.class.php.bak:
```diff
@@ -0,0 +1,60 @@
+<?php
```
Remove this file

bradymiller commented on interface/forms/prior_auth/FormPriorAuth.class.php.bak:
```diff
@@ -0,0 +1,115 @@
+<?php
```
remove this file

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+            var today = document.getElementById("auth_to").value;
+
+            if (EnteredDate > today) {
+                alert("FROM: date is greater than TO: date ");
```
translate and escape with xls function:
``````
alert("<?php echo xls('The FROM date is greater than the TO date'); ?>");

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+			
+            if(document.getElementById("noAuthReq").checked && n === 0){
+			    
+			      alert("Please fill in number of claims");
```
translate and escape with xls function:
``````
alert("<?php echo xls('Please fill in number of claims'); ?>");

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
 </head>
 <body bgcolor="{$STYLE.BGCOLOR2}">
+
 <p><span class="title">Prior Authorization Form</span></p>
```
Lets translate/escape this:
``````
<p><span class="title"><?php echo xlt('Prior Authorization Form'); ?></span></p>

bradymiller replied:
actually, since this is within smarty, can do following instead:
``````
<p><span class="title">{xl t="Prior Authorization Form"}</span></p>

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
 <form name="prior_auth" method="post" action="{$FORM_ACTION}/interface/forms/prior_auth/save.php">
 <table>
 <tr>
-	<td>Prior Authorization Number</td><td><input type="text" size="35" name="prior_auth_number" value="{$prior_auth->get_prior_auth_number()}"></td>
```
Are you using the Prior authorization number anymore?

bradymiller replied:
Need answer to above question

juggernautsei replied:
Yes the prior auth number is being used.

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+
 </tr>
 <tr>
-	<td><br><br>Comments</td>
```
Are you using the comments field anymore?

bradymiller replied:
Need answer to above question

juggernautsei replied:
Yes the comment field is being used.

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
 <table>
 <tr>
-	<td>Prior Authorization Number</td><td><input type="text" size="35" name="prior_auth_number" value="{$prior_auth->get_prior_auth_number()}"></td>
+    <td align="right">No Auth# Req:</td><td><input type="radio" name="not_req" value="Yes" id="noAuthReq" title="Select when No Authorization Number Required" 
```
use xlt()
and remember not to inlcude the : in the trasnlation constant

bradymiller replied:
Actually since in smarty, do following instead:
``````
<td align="right">{xl t="No Auth# Req"}:</td><td><input type="radio" name="not_req" value="Yes" id="noAuthReq" title="{xl t="Select when No Authorization Number Required"}"

juggernautsei replied:
These have been handled per the instruction given.

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
-	<td>Prior Authorization Number</td><td><input type="text" size="35" name="prior_auth_number" value="{$prior_auth->get_prior_auth_number()}"></td>
+    <td align="right">No Auth# Req:</td><td><input type="radio" name="not_req" value="Yes" id="noAuthReq" title="Select when No Authorization Number Required" 
+	       {if $NoAuth == "Yes"} checked {/if}>{if $NoAuth == "Yes"}<br><input type="radio" name="not_req" id="clear_req" value="" 
+		    title="Click here to clear the No Auth need flag" onclick = 'clearThis("claims")'>Clear</td>{/if}
```
xla()

bradymiller replied:
translate via smart {xl t=""}

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+		    title="Click here to clear the No Auth need flag" onclick = 'clearThis("claims")'>Clear</td>{/if}
+</tr>
+<tr>
+    <td align="right">No Auth #Claims:</td><td><input type="text" size="5" name="auth_for" id="claims" value="{$prior_auth->get_auth_for()}" title="Set the number of times no auth can be used">
```
xlt() and xla()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
 </tr>
 <tr>
-	<td colspan="2"><textarea name="comments" value="{$prior_auth->get_comments()}" wrap="virtual" cols="75" rows="8">{$prior_auth->get_comments()}</textarea></td>
+    <td align="right">Description:</td><td><input type="text" size="55" name="desc" value="{$prior_auth->get_desc()}"></td>
```
xlt()

bradymiller replied:
translate via smart {xl t=""}

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+</tr>
+
+<tr>
+    <td align="right">Auth Length:</td><td><label>From: </label>
```
xlt() xlt()

bradymiller replied:
translate via smart {xl t=""}

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+    </td>
+</tr>
+<tr>
+    <td align="right"># Used: </td><td><input type="text" size="5" name="used" value="{$prior_auth->get_used()}" title="Read Only" readonly></td>
```
xlt()

bradymiller replied:
translate via smart {xl t=""}

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+    <td align="right"># Used: </td><td><input type="text" size="5" name="used" value="{$prior_auth->get_used()}" title="Read Only" readonly></td>
+</tr>
+<tr>
+    <td align="right">Auth Contact:</td><td><input type="text" size="25" name="auth_contact" value="{$prior_auth->get_auth_contact()}"></td>
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+    <td align="right">Auth Contact:</td><td><input type="text" size="25" name="auth_contact" value="{$prior_auth->get_auth_contact()}"></td>
+</tr>
+<tr>
+    <td align="right">Auth Phone:</td><td> <input type="text" size="15" name="auth_phone" value="{$prior_auth->get_auth_phone()}">  </td>
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+    <td align="right">Auth Phone:</td><td> <input type="text" size="15" name="auth_phone" value="{$prior_auth->get_auth_phone()}">  </td>
+</tr>
+<tr>
+    <td align="right">Code:</td><td> 
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+	
+</tr>
+<tr>
+	<td align="right">Notes:</td><td colspan="2"><textarea name="comments" value="{$prior_auth->get_comments()}" wrap="virtual" cols="75" rows="8">{$prior_auth->get_comments()}</textarea></td>
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+	<td align="right">Notes:</td><td colspan="2"><textarea name="comments" value="{$prior_auth->get_comments()}" wrap="virtual" cols="75" rows="8">{$prior_auth->get_comments()}</textarea></td>
+</tr>
+<tr>
+	<td align="right">Auth Alert:</td><td><input type="text" size="5" value="" />
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+<tr>
+	<td align="right">Auth Alert:</td><td><input type="text" size="5" value="" />
+        <select><option name="units" value="" /></option>
+		        <option name="units" value="days" {if $Alert == "days"}selected{/if}/>Days</option>
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+	<td align="right">Auth Alert:</td><td><input type="text" size="5" value="" />
+        <select><option name="units" value="" /></option>
+		        <option name="units" value="days" {if $Alert == "days"}selected{/if}/>Days</option>
+                <option name="units" value="units" {if $Alert == "units"}selected{/if}/>Units</option> 
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+        <select><option name="units" value="" /></option>
+		        <option name="units" value="days" {if $Alert == "days"}selected{/if}/>Days</option>
+                <option name="units" value="units" {if $Alert == "units"}selected{/if}/>Units</option> 
+                <option name="units" value="open" {if $Alert == "open"}selected{/if}/>Open Auth</option> 
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+                </select>	</td>
+</tr>
+<tr>
+	<td align="right">Supervisor Override:</td><td><input type="radio" name="override" value="1" {if $OverRide == 1}checked{/if}/>{if $OverRide == 1}<input type="radio" name="override" value="" />Clear{/if}
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html:
```diff
@@ -34,20 +48,134 @@
+	<td align="right">Supervisor Override:</td><td><input type="radio" name="override" value="1" {if $OverRide == 1}checked{/if}/>{if $OverRide == 1}<input type="radio" name="override" value="" />Clear{/if}
+</tr>
+<tr>
+	<td align="right">Archive this auth:</td><td><input type="radio" name="archived" value="1" />
```
xlt()

bradymiller commented on interface/forms/prior_auth/templates/prior_auth/general_new.html.bak:
```diff
@@ -0,0 +1,64 @@
+<html>
```
remove this file

bradymiller commented on library/Claim.class.php.bak:
```diff
@@ -0,0 +1,1309 @@
+<?php
```
remove this file

bradymiller commented:
Review complete.

juggernautsei commented:
Updated the translation for the general form and deleted the bak files. Updated some code changes to better process multiple authorization codes and select the correct one for the CPT that is being billed.

bradymiller commented:
Need a couple answers to above questions(I flagged them above).
What really need to figure out is if this needs to be a new separate form, ie, prior_auth_v2? Will this form be backwards compatible with the old form. For example, is a user has entered data into the old form, and then looks at it on the new form, will it no longer work or will data be lost?

juggernautsei commented:
My apologies, I did not see the questions. However, I think it would be
best to make this v2. It is backwards compatible because the table labels
are in the same name convention. They were not changed. So, if someone were
to use version 2, the previous data would show including comments.

The form does not account for account for upgrading the database to the new
table layout. I had not explored how to make it upgrade able.
The changes to the Claim,class.php file and the gen_x12_837.inc.php files
adds the functionality of the auth being entered into the correct segment
of the 837. That is backwards compatible.

However, I am not sure that the new form can be installed atop of the old
one since they would share the same database table name. It can't be done
from the user interface. Let me know how you would like to proceed.

Sherwin
757-328-2736
Open Med Practice

“For those who believe, no proof is necessary. For those who don't believe,
no proof is possible.” ― Stuart Chase
<http://www.goodreads.com/author/show/1051789.Stuart_Chase>

On Sat, Apr 9, 2016 at 2:44 PM, Brady Miller <notifications@github.com>
wrote:

> Need a couple answers to above questions(I flagged them above).
> What really need to figure out is if this needs to be a new separate form,
> ie, prior_auth_v2? Will this form be backwards compatible with the old
> form. For example, is a user has entered data into the old form, and then
> looks at it on the new form, will it no longer work or will data be lost?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openemr/openemr/pull/109#issuecomment-207829374>
>

bradymiller commented:
Hi Sherwin,
Lets call it something like:
Prior Authorization Advanced Form
prior_auth_advanced

To clean it up:
place it in interface/forms/prior_auth_advanced
revert interface/forms/prior_auth back to original form
remove the stuff you placed in the contrib directory
remove interface/forms/prior_auth/templates/prior_auth/general_new.html.bak
remove interface/forms/prior_auth/FormPriorAuth.class.php.bak

juggernautsei commented:
Brady,
I have made the suggested changes and clean up.

Thanks for working with me.

Sherwin
757-328-2736
Open Med Practice

“For those who believe, no proof is necessary. For those who don't believe,
no proof is possible.” ― Stuart Chase
<http://www.goodreads.com/author/show/1051789.Stuart_Chase>

On Sun, Apr 10, 2016 at 4:57 AM, Brady Miller <notifications@github.com>
wrote:

> Hi Sherwin,
> Lets call it something like:
> Prior Authorization Advanced Form
> prior_auth_advanced
>
> To clean it up:
> place it in interface/forms/prior_auth_advanced
> revert interface/forms/prior_auth back to original form
> remove the stuff you placed in the contrib directory
> remove interface/forms/prior_auth/templates/prior_auth/general_new.html.bak
> remove interface/forms/prior_auth/FormPriorAuth.class.php.bak
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openemr/openemr/pull/109#issuecomment-207947922>
>

robertdown commented:
Closing this since outdated PR
